feature_details.md

```diff
@@ -0,0 +1,174 @@
+
+```
+document.documentTransition.setSameOriginTransitions(
+  {“foo.com/a.html” : [{element : document.html, propertyMap : {“id” : “root”}},
+                                 {element : document.getElementById(‘shared’), propertyMap : {“id” : “shared”}],
+   “foo.com/b.html” : [{element : document.html}]
+  });
+```
+
+* When the new Document loads, the developer can register an `EventListener` to be notified if a transition was initiated by the previous Document. The event, referred to as `handleTransition`, provides the metadata for elements captured from the previous Document and the previous URL, a detailed explanation is at [Referencing Elements](#referencing-elements). The aim of the `handleTransition` event is to allow the developer to define the animations for a transition using information from both Documents. For instance, a paired element transition may be dropped if the corresponding element in the new Document renders offscreen.
+
+* The `handleTransition` event provides a promise based API to declare the transition parameters asynchronously. The motivation to make this asynchronous is to allow the new Document to go through intermediate layouts before it is ready to be displayed. The first paint for the Document is deferred and the previous Document is displayed until this promise is fulfilled and the transition can be started.
+
+```
+document.documentTransition.addEventListener(“handleTransition”, e => {
+    e.initializeTransition(transitionFrom(e.url, e.elements));
+});
+
+async function transitionFrom(previousUrl, elements) {
+    // Wait for resources required for this transition to load.
+    await waitForLoad(previousUrl);
+
+    elementTransitions = []
+    for (let elem in elements) {
+        if (elem.propertyMap[“id”] === “root”) {
+          elementTransitions.push({sourceElement : elem, transitionType: “none”});
+          elementTransitions.push({destinationElement : document.html, transitionType : “fade”});
+        } else if (elem.propertyMap[“id”] === “shared”) {
+          elementTransitions.push({sourceElement : elem, destinationElement : document.getElementById(“shared”));
+        }
+    }
+    return elementTransitions;
+}
+```
+
+* The first rendering lifecycle update after the promise returned by the API resolves is when the animations for this transition are started. This frame also determines the end value for animated properties of shared elements in the new Document, similar to the step on the previous Document to cache the start value for its elements.
+
+* The developer can register another event listener to be notified when all animations associated with this transition have been finished. A caveat to note here is that since the transition is executed with a live DOM for the new Document, any changes to the shared elements in the new Document before this event is dispatched will have undefined behaviour.
```

Comment:
> any changes to the shared elements in the new Document before this event is dispatched will have undefined behaviour.

That doesn't seem ideal. Do you mean that if a shared element has, for example, an animated position via css animations then we can't use that element as a shared target? Why not?